Map Error Status to error tag in Zipkin & Jaeger (#1257)
open-telemetry/opentelemetry-specification: specification/trace/sdk_exporters/jaeger.md

### Status

Span `Status` MUST be reported as a key-value pair in `tags` to Jaeger, unless it is `UNSET`.
In the latter case it MUST NOT be reported.

The following table defines the OpenTelemetry `Status` to Jaeger `tags` mapping.

| Status|Tag Key| Tag Value |
|--|--|--|
|Code | `otel.status_code` | Name of the code, either `OK` or `ERROR`. MUST NOT be set if the code is `UNSET`. |
|Description | `otel.status_description` | Description of the `Status` if it has a value otherwise not set. |

### Error flag

> When Span `Status` is set to `ERROR` an `error` tag SHOULD be added with the

Reviewer: What's the reasoning for using SHOULD instead of MUST here?

Author: My thinking was it works fine without it, you just lose the coloring/icon, so it shouldn't be an absolute type of thing?

Reviewer: I think it should be a MUST, because `error` tag is a very strong signal (coloring is just the UI aspect), and it does not make sense to lose that signal in the conversion when OTEL user explicitly bothered to record the error status.

Author: Updated

> `BOOL` value (`vBool`) set to `true`. The added `error` tag MAY override any

Reviewer: ```suggestion
Boolean value of `true`. The added `error` tag MAY override any
```

The references to `BOOL` and `vBool` are out of place, they are not standard terms, and even in the Jaeger schema they are format-specific.

Author: Good point.
